doc/source/whatsnew/v0.18.1.txt

```diff
@@ -76,8 +76,7 @@ API changes
 
 
 
-
-
+- In ``read_csv``, the Python parser now raises an ``EmptyDataError`` in response to empty columns or header (:issue:`12506`)
```

Comment:
That's because the C engine raises a generic `ValueError` in the analogous place.